File: Medium/1343_number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/Solution.cpp
#include <vector>

class Solution {
public:
    int numOfSubarrays(std::vector<int>& arr, int k, int threshold) {
        int count = 0;
        int sum = 0;

        for (int i = 0; i < k; i++) {
            sum += arr[i];
        }

        if (sum >= threshold * k) {
            count++;
        }

        for (int i = k; i < arr.size(); i++) {
            sum -= arr[i - k];
            sum += arr[i];
            
            if (sum >= threshold * k) {
                count++;
            }
        }

        return count;
    }
};

File: Medium/1343_number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/Solution.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @param {number} threshold
 * @return {number}
 */
var numOfSubarrays = function(arr, k, threshold) {
    let count = 0;
    let sum = 0;

    for (let i = 0; i < k; i++) {
        sum += arr[i];
    }

    if (sum >= threshold * k) {
        count++;
    }

    for (let i = k; i < arr.length; i++) {
        sum -= arr[i - k];
        sum += arr[i];

        if (sum >= threshold * k) {
            count++;
        }
    }

    return count;
};